agent/src/agent/pipeline/pipeline.py

```diff
@@ -448,3 +446,27 @@ class PipelineRetries(Entity):
     def __init__(self, pipeline_: Pipeline):
         self.pipeline_id = pipeline_.name
         self.number_of_error_statuses = 0
+
+
+class PipelineMetric(object):
+    REQUIRED_KEYS = ['counters']
+
+    def __init__(self, metrics_: dict):
+        for key in self.REQUIRED_KEYS:
+            if key not in metrics_:
+                raise KeyError(f'The key `{key}` not in input dict')
+        self.stats = {
+            'in': metrics_['counters']['pipeline.batchInputRecords.counter']['count'],
+            'out': metrics_['counters']['pipeline.batchOutputRecords.counter']['count'],
+            'errors': metrics_['counters']['pipeline.batchErrorRecords.counter']['count'],
+        }
+        self.stats['errors_perc'] = self.stats['errors'] * 100 / self.stats['in'] if self.stats['in'] != 0 else 0
+
+    def __str__(self):
+        return 'In: {in} - Out: {out} - Errors {errors} ({errors_perc:.1f}%)'.format(**self.stats)
+
+    def has_error(self):
+        return self.stats['errors'] > 0
+
+    def has_undelivered(self):
+        return self.stats['out'] == 0 or self.stats['out'] < self.stats['in']
```

Comment:
the situation when there are fewer output metrics than input metrics is completely fine. As an example, a pipeline can filter out some input metrics, or create one Anodot metric from multiple input metrics. So in this situation you can only check if the output is empty